todo_app/data/controller.py

```diff
@@ -0,0 +1,192 @@
+from __future__ import annotations
```

Comment:
The naming of this is slightly confusing; a "controller" typically refers to the class/functions that establish what routes are available, whereas this is the functionality used by your controller - probably more standard would be to refer to this as your `db_service` or similar